client/src/components/Ticketing/ticketingmanager/Season/SeasonInstancesPage.tsx

```diff
@@ -26,45 +26,64 @@ interface SeasonInstancesProp {
 const SeasonInstancesPage = (props: SeasonInstancesProp) => {
   const navigate = useNavigate();
   const {token} = props;
-  const [seasons, setAllSeasons] = useState<Seasons[]>([]);
+  const [filterSetting, setFilterSetting] = useState('active');
+  const [seasonData, setSeasonData] = useState([]);
+  const [allSeasonData, setAllSeasonData] = useState([]);
+  const [inactiveData, setInactiveData] = useState([]);
+  const [activeData, setActiveData] = useState([]);
 
-  const getAllSeasons = async () => {
-    try {
-      const getAllSeasonsRes = await fetch(
-        process.env.REACT_APP_API_2_URL + '/season',
-        {
-          credentials: 'include',
-          method: 'GET',
-          headers: {
-            'Content-Type': 'application/json',
-            'Authorization': `Bearer ${token}`,
-          },
-        },
-      );
+  useEffect(() => {
+    fetch(process.env.REACT_APP_API_2_URL + '/season/list', {
+      headers: {
+        'Authorization': `Bearer ${token}`,
+      },
+    })
+    .then((response) => response.json())
+    .then((data) => {
+      const inactiveSeasons = [];
+      const activeSeasons = [];
+      const allSeasons = [];
 
-      if (!getAllSeasonsRes.ok) {
-        throw new Error('Failed to fetch all seasons');
-      }
+      data.forEach((season) => {
+        const allEventsActive = season.events.every((event) => event.active);
 
-      const seasonsData = await getAllSeasonsRes.json();
-      setAllSeasons(seasonsData);
-    } catch (error) {
-      console.error(error);
-    }
-  };
+        if (allEventsActive) {
+          activeSeasons.push(season);
+        } else {
+          inactiveSeasons.push(season);
+        }
 
-  useEffect(() => {
-    getAllSeasons();
+        // Common operation
+        allSeasons.push(season);
+      });
+      // Sorting
+      inactiveSeasons.sort((a, b) => b.seasonId - a.seasonId);
+      activeSeasons.sort((a, b) => b.seasonId - a.seasonId);
+      allSeasons.sort((a, b) => b.seasonId - a.seasonId);
```

Comment:
It looks like they might be sorted the reverse way, this might be because `seasonId` might be mismatched to what the API returns? Perhaps it should be `seasonid`.

Reply:
thanks, fixed